include/xsimd/math/xsimd_estrin.hpp

```diff
@@ -0,0 +1,103 @@
+/***************************************************************************
+* Copyright (c) Johan Mabille, Sylvain Corlay, Wolf Vollprecht and         *
+* Martin Renou                                                             *
+* Copyright (c) QuantStack                                                 *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XSIMD_ESTRIN_HPP
+#define XSIMD_ESTRIN_HPP
+
+#include "../types/xsimd_types_include.hpp"
+
+namespace xsimd
+{
+    namespace detail
+    {
+        template <class T, uint64_t c, typename T::value_type* = nullptr>
+        inline T coef() noexcept
+        {
+            using value_type = typename T::value_type;
+            return T(caster_t<value_type>(as_unsigned_integer_t<value_type>(c)).f);
+        }
+
+        template <class T, uint64_t c>
+        inline typename std::enable_if<std::is_scalar<T>::value, T>::type coef() noexcept
+        {
+            return T(caster_t<T>(as_unsigned_integer_t<T>(c)).f);
+        }
+
+        template <class T>
+        struct Estrin
```

Comment:
nitpicking: we use snake case, so this should be renamed in `estrin` or `estring_eval` or `estrin_impl`. `estrin` methods defined in the struct can be moved to the private section to improve code readability (and renamed in `eval` for instance if you renamed the struct in `estrin`.

Reply:
Renamed.